lib/sort_by_length.rb

```diff
@@ -1,7 +1,31 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n3), cubic. nested loops, each of them are O(n)
+# Space complexity: O(n), creating new data structure when splitting
 def sort_by_length(my_sentence)
```

Comment:
This works, but it's only O(n^2) and you only need 2 nested loops.  Check out my comments below.